vmod/vlibs/RANDFUNC.vlib

```diff
@@ -129,14 +138,21 @@ input in1;
 reg [(`RANDFUNC_MODNAME_LEN*8)-1:0] modName;
 reg [31:0] hash, arg;
 integer i;
-
+`ifdef VIVADO
+  string s;
+`endif
 begin
 	//	hash = 32'h9e3779b9;
 	hash = 0;
 	// Convert current module hierarchical name to a reg.
 	// Note $sformatf() converts strings right to left.  i.e.
 	// Return value contains data from right most data of the string.
+	`ifdef VIVADO
+	s = $sformatf ("%m");
+	`else
```

Comment:
This seems not quite right (lines 152 and 153 transposed?)

Reply:
You are right, line 152 and 153 should be swopped. I had made the fix in my branch.

Sorry about that.

Reply:
Ok, squashed this commit locally.  Working on batching this up with a handful more changes.